### PR TITLE
Add explanation about the various Requirement types.

### PR DESCRIPTION
**Pull request by johntyree**
Adapted from discussion on slack.

@noraderam do you think this would have been sufficient?

Example here: https://sat-solver.readthedocs.io/en/develop/sat_solving.html

### CODE REVIEW

File: docs/source/sat_solving.rst [hunk 1]
```diff
@@ -45,3 +45,58 @@ to the right of it. ``*`` is a wild-card that matches any version.
...
+There are currently three different ``Requirement`` classes:
+:class:`Requirement<simplesat.constraints.requirement.Requirement>`,
+:class:`InstallRequirement
+<simplesat.constraints.requirement.InstallRequirement>` and
+:class:`ConflictRequirement
+<simplesat.constraints.requirement.ConflictRequirement>`. They have no internal
+differences, but this split allows us to reliably track the origin of a
+requirement via its type and avoid using it in an inappropriate context.
+
+We care about the difference between a requirement created from
```
Comment (noraderam) [inline docs/source/sat_solving.rst]: Who is responsible for creating requirements? Is this done by external code using the solver? If it's done by a particular class or part of the solver process, can you mention that here?

File: docs/source/sat_solving.rst [hunk 2]
```diff
@@ -45,3 +45,58 @@ to the right of it. ``*`` is a wild-card that matches any version.
...
+
+  .. note::
+    Currently, this is the only type of requirement that can be passed to
+    ``modify_requirement``.
+
+:class:`ConflictRequirement <simplesat.constraints.requirement.ConflictRequirement>`
+  Requirements derived from ``package.conflicts`` metadata. For example::
+
+      for constraints in package.conflicts:
+        req = InstallRequirement.from_constraints(constraints)
```
Comment (noraderam) [inline docs/source/sat_solving.rst]: Copypasta--should be `ConflictRequirement`

**Comment by noraderam** [comment]
Otherwise, looks good

File: docs/source/sat_solving.rst (hunk 2 above)
Comment (johntyree) [inline docs/source/sat_solving.rst]: Dang.

File: docs/source/sat_solving.rst (hunk 1 above)
Comment (johntyree) [inline docs/source/sat_solving.rst]: :+1:

Comment (johntyree) [inline docs/source/sat_solving.rst]: Is that better?

**Comment by noraderam** [comment]
LGTM

**Pull request merged**